packages/yarnpkg-pnpify/sources/generateSdk.ts

```diff
@@ -147,168 +262,91 @@ class Wrapper {
   }
 }
 
-export const generateEslintWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const wrapper = new Wrapper(`eslint` as PortablePath, {pnpApi, target});
-
-  await wrapper.writeManifest();
-
-  await wrapper.writeBinary(`bin/eslint.js` as PortablePath);
-  await wrapper.writeFile(`lib/api.js` as PortablePath);
-
-  await addVSCodeWorkspaceSettings(pnpApi, {
-    [`eslint.nodePath`]: npath.fromPortablePath(
-      ppath.dirname(ppath.dirname(ppath.dirname(
-        wrapper.getProjectPathTo(
-          `lib/api.js` as PortablePath,
-        ),
-      ))),
-    ),
-  });
-};
-
-export const generatePrettierWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const wrapper = new Wrapper(`prettier` as PortablePath, {pnpApi, target});
-
-  await wrapper.writeManifest();
-
-  await wrapper.writeBinary(`index.js` as PortablePath);
-
-  await addVSCodeWorkspaceSettings(pnpApi, {
-    [`prettier.prettierPath`]: npath.fromPortablePath(
-      wrapper.getProjectPathTo(
-        `index.js` as PortablePath,
-      ),
-    ),
-  });
-};
-
-export const generateTypescriptLanguageServerWrapper = async (pnpApi: PnpApi, target: PortablePath) => {
-  const wrapper = new Wrapper(`typescript-language-server` as PortablePath, {pnpApi, target});
+export const generateSdk = async (pnpApi: PnpApi, editors: typeof SUPPORTED_EDITORS | null, {report, base}: {report: Report, base: boolean}): Promise<void> => {
+  const topLevelInformation = pnpApi.getPackageInformation(pnpApi.topLevel)!;
+  const projectRoot = npath.toPortablePath(topLevelInformation.packageLocation);
 
-  await wrapper.writeManifest();
+  const targetFolder = ppath.join(projectRoot, PNPIFY_FOLDER);
+  const editorsFilePath = ppath.join(targetFolder, EditorsFile.fileName);
 
-  await wrapper.writeBinary(`lib/cli.js` as PortablePath);
-};
+  const hasEditorsFile = xfs.existsSync(editorsFilePath);
```

Comment:
I feel like the `EditorsFile` class should just take the path to the project root, and derive the path by itself (and would expose a `hasEditorFile` property).